node/src/components/consensus/protocols/highway.rs

```diff
@@ -311,10 +315,23 @@ impl<I: NodeIdT, C: Context + 'static> HighwayProtocol<I, C> {
         if let (Some(value), Some(timestamp), Some(swunit)) =
             (vertex.value(), vertex.timestamp(), vertex.unit())
         {
+            let panorama = &swunit.wire_unit().panorama;
+            let fork_choice = self.highway.state().fork_choice(panorama);
+            let parent_value =
+                fork_choice.map(|hash| self.highway.state().block(hash).value.hash());
+            if timestamp != value.timestamp() || parent_value.as_ref() != value.parent() {
```

Comment:
Note: here we enforce the unit wrapping the proposed candidate block must have the same timestamp as the candidate block.